src/Core/Framework/Struct/JsonSerializableTrait.php

```diff
@@ -9,7 +9,7 @@ public function jsonSerialize(): array
         $vars = get_object_vars($this);
         foreach ($vars as $property => $value) {
             if ($value instanceof \DateTimeInterface) {
-                $value = $value->format(\DateTime::ATOM);
+                $value = $value->format(\DateTime::RFC3339_EXTENDED);
```

Comment:
shouldn't we use the `STORAGE_DATE_TIME_FORMAT` constant from https://github.com/shopware/platform/blob/6.1/src/Core/Defaults.php file?

Reply:
Unfortunately this format does not contain a time zone. `Defaults::STORAGE_DATE_TIME_FORMAT` is the format used to store dates in MySQL/MariaDB. They don't support time zones when storing dates. For the transfer of dates via Json I recommend to use a date format that hat the time zone included.

Reply:
Hey there,

thank you for the pull request! It's intended that we don't store timezone in the database. We use UTC+0 for all time operations in the database and in shopware internally and only convert the timezones when we display them to the user. This prevents a lot of conversion/timezone mismatch errors. Here you can see that we overwrite the PHP timezone setting for each Kernel construct: https://github.com/shopware/platform/blob/6.1/src/Core/Kernel.php#L77

So I think Defaults::STORAGE_DATE_TIME_FORMAT is the way to go.

Reply:
> thank you for the pull request! It's intended that we don't store timezone in the database.

But this is the API, which means you're crossing system boundaries here. As per the API-first approach, we also cannot assume that the client is the Shopware Administration. Because of this, omitting the time zone when returning date/time values from the API seems like a really bad idea, since this will prevent systems interfacing with Shopware from intepreting date/time values unambiguously.

Reply:
There are also classes/entities that implement their own `jsonSerialize()` method if they have some custom serialization logic, if we have agreed to a format to use in json representation please check that it is used consistently

Reply:
Please have a look at the following JS-Snippet. I took it from a browser console that did run on a German PC:

![image](https://user-images.githubusercontent.com/6232639/76745605-72520080-6776-11ea-8df5-0d36e27baf3a.png)

**That means the following:**
If a date-string passed to `new Date()` does not have a time zone it is interpreted with the local time zone of the browser. That means the actual timed that is parsed is dependent on the time zone of the computer that is running the browser. It would then differ from country to country.
If the date-string contains the time zone, it is interpreted as the same time everywhere on the world. Fully independent of the time zone of the browser.
Only in the last case the time transferred via JSON would be consistent on every browser and would match the time actual time saved on the server.
Omitting the time zone in the JSON would lead to many time offset errors.